sklego/meta.py

```diff
@@ -175,3 +174,54 @@ def predict(self, X):
 
     def score(self, X, y):
         return self.estimator_.score(X, y)
+
+
+class Thresholder(BaseEstimator):
+    """
+    Takes a two class estimator and moves the threshold. This way you might
+    design the algorithm to only accept a certain class if the probability
+    for it is larger than, say, 90% instead of 50%.
+    """
+
+    def __init__(self, model, threshold: float):
+        self.model = model
+        self.threshold = threshold
+
+    def _is_classifier(self):
+        return any(['ClassifierMixin' in p.__name__ for p in type(self.model).__bases__])
+
+    def fit(self, X, y):
+        """
+        Fit the data.
+
+        :param X: array-like, shape=(n_columns, n_samples,) training data.
+        :param y: array-like, shape=(n_samples,) training data.
+        :return: Returns an instance of self.
+        """
+        X, y = check_X_y(X, y, estimator=self, dtype=FLOAT_DTYPES)
+        self.estimator_ = clone(self.model)
+        if not self._is_classifier():
```

Comment:
shouldn't this be `if not self.estimator_._is_classifier()`?

Also, we should probably check whether the `estimator` has `predict_proba`. Not all classifiers necessarily have that.

Remember James' trick?

```
class ProbabilisticClassifierMeta(type):
    def __instancecheck__(self, other):
        return hasattr(other, 'predict_proba')

class ProbabilisticClassifier(metaclass=ProbabilisticClassifierMeta):
    pass
    
class SomeModel:
    def predict_proba(self):
        return 1
```

Then in the `fit` you can do:
```
isinstance(estimator, ProbabilisticClassifier)
```
I think it could be neat

Reply:
I've added `sklego.base`.